src/ucs/async/async.c

```diff
@@ -568,6 +582,17 @@ ucs_status_t ucs_async_remove_handler(int id, int is_sync)
         }
     }
 
+    async = handler->async;
+    if (async != NULL) {
+        ucs_mpmc_queue_block(&async->missed);
+        ucs_mpmc_queue_for_each(elem, &async->missed) {
```

Comment:
Need to add `ucs_mpmc_queue_for_each` in `.clang-format` (`ForEachMacros`)

Reply:
maybe instead of exposing ucs_mpmc_queue_lock, ucs_mpmc_queue_unlock and foreach, can add ucs_mpmc_remove_if(mpmc, predicate_cb, arg) to take a callback that returns 0/1 whether the given element should be removed?